src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Assignment.cs

```diff
@@ -14,10 +16,12 @@ public ExprAssignment(Expr lhs, Expr rhs)
             Flags = EXPRFLAG.EXF_ASSGOP;
         }
 
-        public Expr LHS { get; set; }
+        public Expr LHS
+        {
+            get => _lhs;
+            set => Type = (_lhs = value).Type;
```

Comment:
Looks like it is written to in VisitChildren

---
In reply to: [130415436](https://github.com/dotnet/corefx/pull/22684#discussion_r130415436) [](ancestors = 130415436,129837030)